modules/cold-extraction/ColdDataRetriever.py

```diff
@@ -162,22 +162,28 @@ def initialize():
     logging.info("Number of running Niffler processes: {0} and storescp processes: {1}".format(niffler_processes,
                                                                                                storescp_processes))
 
-    if niffler_processes > MAX_PROCESSES:
-        logging.error("[EXTRACTION FAILURE] {0}: Previous {1} extractions still running. As such, your extraction "
-                      "attempt was not successful this time. Please wait until those complete and re-run your"
-                      " query.".format(datetime.datetime.now(), MAX_PROCESSES))
-        sys.exit(0)
+    query_aet = QUERY_AET.split(":")[0]
+    dest_aet = DEST_AET
+    if query_aet == dest_aet:
+        logging.info("{0}: StoreScp process for the current Niffler extraction is starting now".format(
+            datetime.datetime.now()))
 
-    if storescp_processes >= niffler_processes:
-        logging.info('Killing the idling storescp processes')       
-        check_kill_process()
+        if not file_path == cfind_only and not file_path == cfind_detailed:
+            subprocess.call("{0}/storescp --accept-unknown --directory {1} --filepath {2} -b {3} > storescp.out &".format(
+                DCM4CHE_BIN, storage_folder, file_path, QUERY_AET), shell=True)
 
-    logging.info("{0}: StoreScp process for the current Niffler extraction is starting now".format(
-        datetime.datetime.now()))
+    else:
```

Comment:
You have broken the whole code block and reorganized things, which was not needed.

It was meant to be more like a conversion of CODE-BLOCK -> if-condition {CODE-BLOCK}, else.

Please read your code again.

The checks and the following code blocks of "if storescp_processes >= niffler_processes:" and "if not file_path == cfind_only and not file_path == cfind_detailed:" should be under "if query_aet == dest_aet:" rather than under "else."